cmds/wifi/wifi.go

```diff
@@ -44,6 +43,11 @@ func main() {
 		conf  []byte
 	)
 
+	if flag.NArg() < 1 {
+		flag.Usage()
+		os.Exit(1)
+	}
+
 	flag.Parse()
 	a := flag.Args()
 	essid = a[0]
```

Comment:
by far the easiest way to fix this:note how we assign things in the case. Just move the essid assignment into the case. Then you'll fall into the default if len(a) is not a valid value, including 0. This turns into +2 lines, but you don't need the check for flag.Narg then.

the essid assign dates from the very earliest version of this program,.

so
case len(a) == 3:
essid = a[0]
 conf = a[0], []byte etc. etc. 
case len(1) == 2:
essid, pass = a[0], a[1]
...
case len(a) == 1:
essid = a[0]
conf = etc. etc.

sure, duplicate assign, but it still reads cleaner to me, as there is one error handling case, the 
default:

Reply:
All done!